includes/class-wc-subscriptions-synchroniser.php

```diff
@@ -686,8 +686,10 @@ public static function calculate_first_payment_date( $product, $type = 'mysql',
 					$month_number = gmdate( 'm', wcs_add_months( $from_timestamp, $interval ) );
 				}
 			}
+			$number_of_months = WC_Subscriptions_Product::get_interval( $product );
```

Comment:
> The reason is because $interval comes with the logic around whether the customer is being charged upfront. ie because the customer is being charged upfront, the customer is entitled to the billing interval in part or in full. If they aren't being charged upfront, they are paying either $0 or a prorated amount and so should be given the next possible payment date available (ie don't add a year).

Thanks for the explanation @james-allan. This did not come to my mind.
I have addressed this in [d9b4f21](https://github.com/Automattic/woocommerce-subscriptions-core/pull/651/commits/d9b4f21a5515f22a51fec63709ab275008572c54)